### DISCUSSION TITLE
Feature Request - Reverse Constructor Order

### DISCUSSION THREAD

**Discussion by moonymachine** [dotnet/docfx#8661] [Ideas]
When a class has multiple constructors with a varying numbers of parameters, the order of the constructors in the api documentation begins with the parameterless constructor, but then displays from most to least parameters, opposite of how they are listed in the source code.

Example:
```csharp
public class MyClass
{
	public int Foo { get; }
	public char Bar { get; }
	public string Baz { get; }

	public MyClass() : this(0, '\0', string.Empty) { }

	public MyClass(int foo) : this(foo, '\0', string.Empty) { }

	public MyClass(int foo, char bar) : this(foo, bar, string.Empty) { }

	public MyClass(int foo, char bar, string baz)
	{
		Foo = foo;
		Bar = bar;
		Baz = baz;
	}
}
```

Currently, these constructors will be displayed in the following order in the resulting docs:

MyClass()
MyClass(int, char, string)
MyClass(int, char)
MyClass(int)

**Comment by yufeih** [comment]
Looks like a sorting bug. Methods with more parameters should appear last.

**Reply by moonymachine** [reply]
As I'm looking through my API reference more today, I notice that things just are not in the same order they are in the source code. I'm not sure if it would be difficult, or if it is even possible, but I would love to have the docs match the order of how my members are listed in source. Obviously it would not be exactly the same, as fields, properties, methods, etc are grouped by category. If it's not possible, or prohibitively difficult, that's understandable, but I sure would love to see it.

**Reply by moonymachine** [reply]
I updated docfx to version 2.66 yesterday, and now the constructors are listed in the expected order.

(I'm not sure exactly which version because I download the zip for a specific version to a local directory relative to this particular documentation project, and when I try to run `docfx help` `docfx help -a` or `docfx --help` I just get an error. The docfx.exe has no version listed in the properties.)
